src/platform/site-wide/user-nav/containers/AutoSSO.jsx

```diff
@@ -0,0 +1,55 @@
+import { connect } from 'react-redux';
+
+import { checkKeepAlive } from 'platform/user/authentication/actions';
+import {
+  ssoe,
+  ssoeInbound,
+  hasCheckedKeepAlive,
+} from 'platform/user/authentication/selectors';
+import { autoLogin, autoLogout } from 'platform/user/authentication/utilities';
+import { hasSession, hasSessionSSO } from 'platform/user/profile/utilities';
+import environment from 'platform/utilities/environment';
+import { ssoKeepAliveSession } from 'platform/utilities/sso';
+
+export async function checkStatus(toggleKeepAlive) {
+  await ssoKeepAliveSession();
+  if (hasSession() && !hasSessionSSO()) {
+    autoLogout();
+  } else if (!hasSession() && hasSessionSSO()) {
+    autoLogin();
+  }
+
+  toggleKeepAlive();
+}
+
+function AutoSSO(props) {
+  const { useSSOe, useInboundSSOe, hasCalledKeepAlive } = props;
+
+  if (
+    useSSOe &&
```

Comment:
Just brainstorming, but `useSSOe` could possibly return false each time if `environment.isLocalhost()`

Reply:
that's a good point. we want that to happen every time, so might as well put it in the selector.

note: it's `useInboundSSOe` we want that condition for. `useSSOe` is fine for localhost